library/core/src/ptr/const_ptr.rs

```diff
@@ -906,6 +957,55 @@ impl<T> *const [T] {
         // SAFETY: the caller ensures that `self` is dereferencable and `index` in-bounds.
         unsafe { index.get_unchecked(self) }
     }
+
+    /// Returns `None` if the pointer is null, or else returns a shared slice to
+    /// the value wrapped in `Some`. In contrast to [`as_ref`], this does not require
+    /// that the value has to be initialized.
+    ///
+    /// [`as_ref`]: #method.as_ref
+    ///
+    /// # Safety
+    ///
+    /// When calling this method, you have to ensure that *either* the pointer is NULL *or*
+    /// all of the following is true:
+    ///
+    /// * The pointer must be [valid] for reads for `ptr.len() * mem::size_of::<T>()` many bytes,
+    ///   and it must be properly aligned. This means in particular:
+    ///
+    ///     * The entire memory range of this slice must be contained within a single allocated object!
+    ///       Slices can never span across multiple allocated objects, and
+    ///
+    ///     * the pointer must be aligned even for zero-length slices. One
```

Comment:
Still looks like such an odd sentence here though?

Reply:
For me it felt natural to continue the sentence there, since both complement the previous sentence, but I think that's a matter of taste. If you want, I can change that.

Reply:
The previous sentence here is "The entire memory range of this slice must be contained within a single allocated object!". I don't see how the second bullet point complements that point.

The two bullets are two separate independent points, and yet somehow the second half of the first bullet and the beginning of the second bullet are glued into one sentence. I don't see how that makes any sense.

Reply:
Maybe you mean that both complement "The pointer must be valid for reads", but that is not what the grammatical structure indicates here. The sentence structure indicates a connection between "must be contained within a single allocated object" on the one hand, and ("never span across multiple allocated objects" + "must be aligned even for zero-length slices") on the other hand, which to me just makes no sense.

Reply:
Well, I think you are right. Reinvestigating [`slice::from_raw_parts`](https://doc.rust-lang.org/nightly/core/slice/fn.from_raw_parts.html) shows, that its documentation also uses two separate sentences.

Reply:
Do you think it is enough if I change it as follows, or should it be wound up a bit more?
```suggestion
    ///       Slices can never span across multiple allocated objects.
    ///
    ///     * The pointer must be aligned even for zero-length slices. One
```

Reply:
That diff seems fine to me. What other changes were you thinking of?

Reply:
I do not have any concrete ones in mind at the moment, maybe you can think of something else?

Reply:
Not really, that's why I am asking. ;)
Let's just go with that diff.